Source/objects.cpp

```diff
@@ -3978,10 +3981,10 @@ void OperateShrine(int pnum, int i, int sType)
 			return;
 		InitDiabloMsg(EMSG_SHRINE_GLOWING);
 		int playerXP = plr[myplr]._pExperience;
-		int xpLoss, magicGain;
+		Sint32 xpLoss, magicGain;

```

Comment:
Isn't this supposed to be `Sint64`?

Reply:
PlayerXP is singed 32bit int so multiplied by 0.95 should still fit in a Sint32 variable. The original __int64 casting appears mostly redundant.